tests/DCPS/Compiler/xcdr/xcdr.idl

```diff
@@ -72,6 +72,27 @@ struct MutableXcdr12Struct {
   COMMON_FIELDS
 };
 
+enum CommonFields {
+  E_SHORT_FIELD,
+  E_LONG_FIELD,
+  E_OCTET_FIELD,
+  E_LONG_LONG_FIELD
+};
+
+@mutable
+@OpenDDS::data_representation(XCDR1)
+@OpenDDS::data_representation(XCDR2)
+union MutableXcdr12Union switch (CommonFields) {
+case E_SHORT_FIELD:
+  short short_field;
+case E_LONG_FIELD:
+  long long_field;
+case E_OCTET_FIELD:
+  octet octet_field;
+case E_LONG_LONG_FIELD:
+  long long long_long_field;
+};
```

Comment:
This looks quite similar to the `MutableUnion` below. Their tests also look similar. Do we need both?

Reply:
The difference is that IDs are defaulted in one and explicitly defined in the other. I think structure tests use both, and I tried to parallel unions to them.

Reply:
I think explicit IDs are mostly used for testing mutable structs with reordered members or additional members in the middle of existing members.